File: reverse_integer.cpp
# Reverse_integer
#include<iostream>
using namespace std;

class Solution 
{
public:
	int reverse(int x)
	{
		long long y = 0;
		while (x)     //the point
		{
			y = y * 10 + x % 10;
			x = x / 10;
		}
		if ((y > INT_MAX) || (y < INT_MIN))
			return 0;
		else
			return y;
	}
};

void main()
{
	Solution s;
	s.reverse(1534236469);
}

File: reverse_integer.java
package reverse_integer;

public class solution {

	public int reverse(int x)
	{
		int y=0;
		int z=0;
		while(x!=0)
		{
			z=y*10+x%10;
			if((z-x%10)/10!=y)   //if overflow,the z will not equals to y
				return 0;
			else
			{
				y=z;
			}
			x=x/10;
		}
		System.out.println(y);
			return y;
	}
}
